src/util/evalUnits.ts

```diff
@@ -13,36 +13,35 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import BN from 'bn.js';
+
+const getSiValue = (si: number): BN => new BN(10).pow(new BN(si));
+
 const si = [
-  { value: -1e24, symbol: 'y' },
-  { value: -1e21, symbol: 'z' },
-  { value: -1e18, symbol: 'a' },
-  { value: -1e15, symbol: 'f' },
-  { value: -1e12, symbol: 'p' },
-  { value: -1e9, symbol: 'n' },
-  { value: -1e6, symbol: 'μ' },
-  { value: -1e3, symbol: 'm' },
-  { value: 1, symbol: '' },
-  { value: 1e3, symbol: 'k' },
-  { value: 1e6, symbol: 'M' },
-  { value: 1e9, symbol: 'G' },
-  { value: 1e12, symbol: 'T' },
-  { value: 1e15, symbol: 'P' },
-  { value: 1e18, symbol: 'E' },
-  { value: 1e21, symbol: 'Z' },
-  { value: 1e24, symbol: 'Y' }
+  { value: new BN(1), symbol: '' },
```

Comment:
Hmm, I think it would be cool to support milli units. I know it's not possible to store in `BN` type, but perhaps we could have something like: `{ negExp: true, exp: 3 }` and convert to BN only after we multiply by decimals (in case of `negExp` you divide by `new BN(10).pow(new BN(exp))` in case of `negExp == false` you multiply).

Reply:
I have implemented this as per your suggestion with a twist - meaning that first I multiply (in every case) with chainDecimals, and then div (in case of using neg) with the siVal.value or (in case of normal) multiply with the siVal.value.
There is still though a chance that there will be loss on conversion.
For example if the user types in `0.0000035m` that will actually send `3` or in another scenario if user types `1000000.3f` then this will send `1`.

What can be done in order to "catch" this is to create a string with calculation in floats and BN, compare the outcome and if it is the same, then send it along (but then again - we may gave conversion issues with the floats - right?)

What do you thing @tomusdrw ?

Reply:
I think the cases you've mentioned are fine since that's simply below what is possible to be sent (i.e. it's not our mistake of parsing that incorrectly - it's just the user is trying to send a value that is impossible to send).

I think with payload/amount confirmation it should be fine to ignore that edge case. If you really want to, we could also display a warning by doing something like:
```js
const number = ...;
if (number !== number.div(siValue.value).mul(siValue.value)) {
  return "Warning: truncated some of the value to transfer because it's below one planck." 
}
```